src/marketplace/MakeAComponentSaleView.js

```diff
@@ -68,15 +68,26 @@ class MakeAComponentSaleView extends React.Component<Props, State> {
 	};
 
 	makeASaleOfAComponent =  ():void => {
+		//Catch if we are not selling any amount or if selling more than what we have
+		const indexOfItem = this.getIndexOfListingArray(this.state.itemID);
+		if(this.state.itemAmount <= 0 || this.state.itemAmount > this.state.itemsToSell[indexOfItem].amount) {
```

Comment:
This error catching needs to happen on the backend. If the server isn't returning an error, we have a bug there and we want to know about it. We don't want to do validation in two places because that doubles the amount of code we have to look through if something is wrong.